rcore/flash.c

```diff
@@ -158,54 +189,58 @@ static void _fs_init()
                 KERN_LOG("flash", APP_LOG_LEVEL_ERROR, "this filesystem has a blank page in it... hmm...");
             saw_blank_page = 1;
             continue;
         }
-        if (hdr.v_0x5001 != 0x5001) {
-            KERN_LOG("flash", APP_LOG_LEVEL_ERROR, "page %d has bad header version 0x%04x; I give up", pg, hdr.v_0x5001);
+        if (hdr->v_0x5001 != 0x5001) {
+            KERN_LOG("flash", APP_LOG_LEVEL_ERROR, "page %d has bad header version 0x%04x; I give up", pg, hdr->v_0x5001);
             _fs_valid = 0;
             return;
         }
-        if (hdr.status == 0xFE && lastpg != -1) {
+        if (hdr->status == 0xFE && lastpg != -1) {
             lastpg = pg;
         }
-        if ((lastpg != -1) && FLASHFLAG(hdr.empty, HDR_EMPTY_ALLOCATED)) {
+        if ((lastpg != -1) && FLASHFLAG(hdr->empty, HDR_EMPTY_ALLOCATED)) {
             if (!saw_page_in_outer_space)
                 KERN_LOG("flash", APP_LOG_LEVEL_ERROR, "page %d is marked as allocated, but page %d had the last page marker... hmm...", pg, lastpg);
             saw_page_in_outer_space = 1;
         }
         
         /* The rest of the checks only apply to an allocated page. */
-        if (!FLASHFLAG(hdr.empty, HDR_EMPTY_ALLOCATED))
+        if (!FLASHFLAG(hdr->empty, HDR_EMPTY_ALLOCATED))
             continue;
-        if (!FLASHFLAG(hdr.status, HDR_STATUS_FILE_START))
+
+        if (FLASHFLAG(hdr->status, HDR_STATUS_FILE_CONT))
+            _fs_set_page_state(pg, PageStateFileCont);
+
+        if (!FLASHFLAG(hdr->status, HDR_STATUS_FILE_START))
             continue;
-        if (!FLASHFLAG(hdr.status, HDR_STATUS_DEAD) && hdr.st_create_complete) {
+
+        if (!FLASHFLAG(hdr->status, HDR_STATUS_DEAD) && hdr->st_create_complete) {
             KERN_LOG("flash", APP_LOG_LEVEL_ERROR, "page %d creation not complete; I can't deal with this; go boot PebbleOS to clean up first", pg);
             _fs_valid = 0;
             return;
         }
-        if (FLASHFLAG(hdr.status, HDR_STATUS_DEAD) && hdr.st_delete_complete) {
+        if (FLASHFLAG(hdr->status, HDR_STATUS_DEAD) && hdr->st_delete_complete) {
             KERN_LOG("flash", APP_LOG_LEVEL_ERROR, "page %d deletion not complete; I can't deal with this; go boot PebbleOS to clean up first", pg);
             _fs_valid = 0;
             return;
         }
-        
-        if (hdr.filename_len == 2) {
-            /* Uh oh. */
-            char c[3];
-            c[2] = 0;
-            _fs_read_page_ofs(pg, sizeof(struct page_hdr), c, 2);
-            if (!strcmp(c, "GC")) {
-                KERN_LOG("flash", APP_LOG_LEVEL_ERROR, "page %d has a GC file; I can't deal with this; go boot PebbleOS to clean up first", pg);
-                _fs_valid = 0;
-                return;
-            }
+
+        if (hdr->filename_len > MAX_FILENAME_LEN)
+            KERN_LOG("flash", APP_LOG_LEVEL_ERROR, "page %d has unexpectedly long file name: %d; it may cause further errors", pg, hdr->filename_len);
```

Comment:
only loggin here, not setting to invalid, as this is not FS error, only wrong assumptions on my side, this will not crash, such file will just not be found by `fs_find_file` 

I'm no longer sure if this optimization with reading a name in one go is worth it, considering this possible bug. Should I remove that and just read filename separately (would require allocation here)?